heat/core/tests/test_rounding.py

```diff
@@ -162,6 +162,70 @@ def test_floor(self):
         with self.assertRaises(TypeError):
             ht.floor(object())
 
+    def test_modf(self):
+        start, end, step = -5.0, 5.0, 1.4
+        comparison = np.modf(np.arange(start, end, step, np.float32))
+
+        # exponential of float32
+        float32_tensor = ht.arange(start, end, step, dtype=ht.float32)
+        float32_modf = float32_tensor.modf()
+        self.assertIsInstance(float32_modf[0], ht.DNDarray)
+        self.assertIsInstance(float32_modf[1], ht.DNDarray)
+        self.assertEqual(float32_modf[0].dtype, ht.float32)
+        self.assertEqual(float32_modf[1].dtype, ht.float32)
+        self.assertTrue((x for x in float32_modf[0]._DNDarray__array) == y for y in comparison[0])
+        self.assertTrue((x for x in float32_modf[1]._DNDarray__array) == y for y in comparison[1])
+
+        # exponential of float64
+        comparison = np.modf(np.arange(start, end, step, np.float64))
+
+        float64_tensor = ht.arange(start, end, step, dtype=ht.float64)
+        float64_modf = float64_tensor.modf()
+        self.assertIsInstance(float64_modf[0], ht.DNDarray)
+        self.assertIsInstance(float64_modf[1], ht.DNDarray)
+        self.assertEqual(float64_modf[0].dtype, ht.float64)
+        self.assertEqual(float64_modf[1].dtype, ht.float64)
+        self.assertTrue((x for x in float32_modf[0]._DNDarray__array) == y for y in comparison[0])
```

Comment:
while technically correct, a neater way to do this might be to use `ht.all(tensor == ht.array(comparison))` or `torch.all` as you use in the test_round function